pynamodb/expressions/update.py

```diff
@@ -0,0 +1,161 @@
+from pynamodb.constants import BINARY_SET_SHORT, LIST_SHORT, NUMBER_SET_SHORT, NUMBER_SHORT, STRING_SET_SHORT
+from pynamodb.expressions.util import get_value_placeholder, substitute_names
+
+
+class Action(object):
+    format_string = ''
+
+    def __init__(self, path, value=None):
+        self.path = path
+        self.value = value
+
+    def serialize(self, placeholder_names, expression_attribute_values):
+        path = substitute_names(self.path, placeholder_names, split=True)
+        value = get_value_placeholder(self.value, expression_attribute_values) if self.value else None
+        return self.format_string.format(value, path=path)
+
+
+class SetAction(Action):
+    """
+    The SET action adds an attribute to an item.
+    """
+    format_string = '{path} = {0}'
+
+    def __init__(self, path, value):
+        super(SetAction, self).__init__(path, value)
+
+
+class IncrementAction(SetAction):
+    """
+    A SET action that is used to add to a number attribute.
+    """
+    format_string = '{path} = {path} + {0}'
+
+    def __init__(self, path, amount):
+        (attr_type, value), = amount.items()
+        if attr_type != NUMBER_SHORT:
+            raise ValueError("{0} must be a number".format(value))
+        super(IncrementAction, self).__init__(path, amount)
+
+
+class DecrementAction(SetAction):
+    """
+    A SET action that is used to subtract from a number attribute.
+    """
+    format_string = '{path} = {path} - {0}'
+
+    def __init__(self, path, amount):
+        (attr_type, value), = amount.items()
+        if attr_type != NUMBER_SHORT:
+            raise ValueError("{0} must be a number".format(value))
+        super(DecrementAction, self).__init__(path, amount)
+
+
+class AppendAction(SetAction):
+    """
+    A SET action that appends elements to the end of a list.
+    """
+    format_string = '{path} = list_append({path}, {0})'
```

Comment:
is `list_append` a dynamo primative / op?

Reply:
it is

Reply:
yeah -- i don't use it here yet but i thought it easiest to add all the allowed values